examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
```

Comment:
I guess this is a matter of taste. I do not like to clutter code with ternary statements as it might make porting changes between the different sim_fibo_ads more involved. But if you insist...